examples/language/lib/classes/shorthand.dart

```diff
@@ -0,0 +1,172 @@
+// #docregion intro
+// Enum example
+enum Status { none, running, stopped, paused }
+Status currentStatus = .running; // Instead of Status.running
+
+// Static method example
+int port = .parse('8080'); // Instead of int.parse('8080')
+
+// Constructor example
+class Point {
+  final int x, y;
+  Point(this.x, this.y);
+  Point.origin() : x = 0, y = 0;
+}
+Point origin = .origin(); // Instead of Point.origin()
+// #enddocregion intro
+
+// #docregion enums
+// Enum example
+enum LogLevel { debug, info, warning, error }
+
+// Function to get a color code based on log level
+String colorCode(LogLevel level) {
+  return switch (level) {
+    .debug   => 'gray',    // Instead of LogLevel.debug
+    .info    => 'blue',    // Instead of LogLevel.info
+    .warning => 'orange',  // Instead of LogLevel.warning
+    .error   => 'red',     // Instead of LogLevel.error
+  };
+}
+
+// Example usage:
+var warnColor = colorCode(.warning); // Returns 'orange'
+
+// #enddocregion enums
+
+// #docregion methods
+// Static method
+int httpPort = .parse('80'); // Instead of int.parse('80')
+
+// Static field/getter
+BigInt bigIntZero = .zero; // Instead of BigInt.zero
+// #enddocregion methods
+ 
+// #docregion constructors
+ class Point {
+  final double x, y;
+  const Point(this.x, this.y);
+  const Point.origin() : x = 0, y = 0; // Named constructor
+
+  // Factory constructor
+  factory Point.fromList(List<double> list) {
+    return Point(list[0], list[1]);
+  }
+}
+
+// Named constructor
+Point origin = .origin(); // Instead of Point.origin()
+
+// Factory constructor
+Point p1 = .fromList([1.0, 2.0]); // Instead of Point.fromList([1.0, 2.0])
+
+// Generic class constructor
+List<int> intList = .filled(5, 0); // Instead of List.filled(5, 0)
+// #enddocregion constructors
+
+// #docregion tearoff
+
+// #enddocregion tearoff
+
+// #docregion chain
+// .fromCharCode(72) resolves to the String "H",
+// then the instance method .toLowerCase() is called on that String.
+String lowerH = .fromCharCode(72).toLowerCase(); 
+// Instead of String.fromCharCode(72).toLowerCase()
+
+print(lowerH); // Output: h
+// #enddocregion chain
+
+// #docregion allowedequality
```

Comment:
When naming docregions with multiple words, we have a convention of using dashes between the word. So in this case, `allowed-equality`.